### ISSUE TITLE
Improve translation support/PO file support of "special" messages in %Lexicon

### ISSUE THREAD

**Issue by taniwallach** [openwebwork/webwork2#1079]
This is pasted from an old email discussion:

The `_GUEST_LOGIN_MESSAGE`  seems to be a special placeholder, and may have been messed up by the way I made the updated POT and PO files. The problem is also a bug in the revised PO files for 2.14.

For 2.14, I think we can probably just manually fix the English PO files (maybe also the POT file) to include the expected English strings for these placeholder messages. That should suffice to get 2.14 out the door with minimal additional changes and effort. What do you think? **Was done**

However, I think the root cause is a bigger problem.

Line 285 of `webwork2/lib/WeBWorK/ContentGenerator/Login.pm` has
```
print CGI::p($r->maketext("_GUEST_LOGIN_MESSAGE", CGI::b($r->maketext("Guest Login"))));
```

`webwork2/lib/WeBWorK/Localize.pm` has a strings for this message and several message like this set inside a special hash called `%Lexicon`.

Transifex does not have the intended English string value for the `_GUEST_LOGIN_MESSAGE` in the English file, nor did the updated POT file generated from the 2.14 codebase using `xgettext.pl` running inside a WW Docker container.

I suspect after a quick look that this is a deeper problem with the way Transifex handles the source language, and the way `xgettext.pl` works and how those things interact with the way WW tried to use these placeholder codes defined in the `%Lexicon` hash in `webwork2/lib/WeBWorK/Localize.pm`.

From what I see in Transifex quickly, you cannot edit "translations" in the source language, so using a "placeholder" like `_GUEST_LOGIN_MESSAGE` cannot work with Transifex. It could be that if the `msgstr` in the `en_us.po` file was removed than Perl's `maketext` will somehow get the right English string from the `%Lexicon`, but I really don't know if that is supposed to or will happen.

I think that it is probably necessary to get rid of the placeholder codes totally and just replace the placeholder codes with the real English message text in the Perl files. It will then be necessary to fix up the POT and PO files. Then translations in other languages can be made for the "real" strings. 

Doing this, unless a better approach is found, should probably be left for devel and not 2.14.

There are 14 hash keys in `%Lexicon`, 9 of which have entries in the PO/POT files and all 9 have the problem that the placeholder string appears instead of the "real" string as both `msgid` and `msgstr` produced by Transifex. Another 3 appear commented out in the PO files (`_REDUCED_CREDIT_MESSAGE_1`, `_REDUCED_CREDIT_MESSAGE_2`, `_AUTO`) so apparently are no longer in use. 

The final two entries in `%Lexicon` are `_PERMISSIONS`  and `_STATUS` which are references to **arrays** of strings, and seem even more special, as they seem to define the various permission levels and status codes. If these are internal codes used as values in the databases, I'm not sure there is any reason to translate them.

**Comment by taniwallach** [comment]
@mgage responded to the email quoted above:

> Date: Mon, 22 Oct 2018 20:51:15 +0000
> From: Michael Gage
>
> I’ll have to think about this some more — it’s clear that you’ve identified a real problem.  We got tired of translating word for word things like the  ‘guest login message’ and simply put in place holders that would generate the appropriate greeting in which ever language.  This was long before transifex came along so we weren’t making allowances for how this interact with this use of .pot and .po files.
>
> I agree that solving the details of this should wait for devel and not 2.14.  I’m hoping for a clever solution that will duplicate what we had originally and still work with Transifex.

**Comment by taniwallach** [comment]
While thinking about some translation related emails, I recalled this issue and opened it here.

As noted above, we needed to manually fix the English PO files merged from Transifex into a new phrase file in order to to set values and override what Transifex provided for English for these special phrases.

I thought of 2 options to try to solve this for the future.

One option is just to drop the use of `%Lexicon` and put the full English strings directly in place, so the maketext/Transifex systems will have them in the PO/POT files as a phrase and not a "code name".

The second option I came up with is to have such messages be handled by a 2 stage process in the code.

Stage 1. Use the maketext call and store the result in a variable. For non-English languages the result we want would be a real translation, but for English we would expect just the "ID" back (ex.
`_GUEST_LOGIN_MESSAGE`).

Stage 2. If the result of the maketext call is the "original" string sent to maketext, then use the `%Lexicon` based replacement to get the "long" English string referred to by the ID.

The second option seems more in line with what Mike had hoped could be developed.
